code/modules/surgery/organs/stomach/external/wings.dm

```diff
@@ -0,0 +1,203 @@
+///Wing base type. doesn't really do anything
+/obj/item/organ/external/wings
+	zone = BODY_ZONE_CHEST
+	slot = ORGAN_SLOT_EXTERNAL_WINGS
+	layers = list(EXTERNAL_BEHIND, EXTERNAL_ADJACENT, EXTERNAL_FRONT)
+
+	preference = "wings"
+
+/obj/item/organ/external/wings/can_draw_on_bodypart(mob/living/carbon/human/human)
+	return !( (human.wear_suit?.flags_inv & HIDEJUMPSUIT) && !(human.wear_suit.species_exception && is_type_in_list(src, human.wear_suit.species_exception) ) )
+
+///The true wings that you can use to fly and shit (you cant actually shit with them, but it does wing stuff)
+/obj/item/organ/external/wings/functional
+	///The flight action object
+	var/datum/action/innate/flight/fly
+
+	///The preference type for opened wings
+	var/wings_open_preference = "wingsopen"
+	///The preference type for closed wings
+	var/wings_closed_preference = "wings"
+
+	///Are our wings open or closed?
+	var/wings_open = FALSE
+
+/obj/item/organ/external/wings/functional/get_global_feature_list()
+	if(wings_open)
+		return GLOB.wings_open_list
+	else
+		return GLOB.wings_list
+
+/obj/item/organ/external/wings/functional/Insert(mob/living/carbon/reciever, special, drop_if_replaced)
+	. = ..()
+
+	if(isnull(fly))
+		fly = new
+		fly.Grant(reciever)
+
+/obj/item/organ/external/wings/functional/Remove(mob/living/carbon/organ_owner, special)
+	. = ..()
+
+	fly.Remove(organ_owner)
+
+/obj/item/organ/external/wings/functional/on_life(delta_time, times_fired)
+	. = ..()
+
+	HandleFlight(owner)
+
+///Mostly as a starter to check if we can still fly
+/obj/item/organ/external/wings/functional/proc/HandleFlight(mob/living/carbon/human/H)
```

Comment:
https://github.com/tgstation/tgstation/blob/master/.github/CONTRIBUTING.md#use-descriptive-and-obvious-names

Reply:
Also, name it `handle_flight`